arbor/backends/gpu/shared_state.cpp

```diff
@@ -274,6 +306,18 @@ const arb_value_type* shared_state::mechanism_state_data(const mechanism& m, con
     return nullptr;
 }
 
+const arb_value_type* shared_state::mechanism_prng_state_data(const mechanism& m, const std::string& key, unsigned cache_index) {
+    const auto& store = storage.at(m.mechanism_id());
+
+    if (cache_index >= random_number_cache_size) return nullptr;
```

Comment:
well, this function is used by the probes. I returned a `nullptr` because that's what has been done for the state probes above...

Reply:
Yes, and that's ok for the 'we didn't find the item you're looking for' case. Here it seemed like the internal
state of the cache is broken. But this can happen if the internal pre-fetched prng cache is exhausted, right?
Question is, does this imply the same response as 'no, we do not know this variable'? I feel it should be different,
maybe return the last state instead? Or a different tag.

Reply:
I think you've misunderstood. The internal state of the cache is not broken, and cannot break due to the probe asking. The probe asks for a cache line with a certain index (`cache_index` - a user supplied argument), and gets it if it exists. It exists if `cache_index` is smaller than  the number of caches (`random_number_cache_size`), which is a constant. So I'd argue it is exactly the same case as you mention.
In terms of elegance, I agree it is not very nice. It could go away, If I had a good way to probe in the tests without actually making probes...

Reply:
probes are gone, and so is this function